dali/base/sysinfologger.cpp

```diff
@@ -0,0 +1,891 @@
+    }
+    virtual bool first() override
+    {
+        StringBuffer xpath(SYS_INFO_ROOT);
+        if(filter->queryMatchYear() && filter->queryMatchMonth())
+        {
+            xpath.appendf("/m%04u%02u", filter->queryMatchYear(), filter->queryMatchMonth());
+            if (filter->queryMatchDay())
+                xpath.appendf("/d%02u", filter->queryMatchDay());
+        }
+        unsigned mode = RTM_LOCK_READ | (updateable?RTM_LOCK_WRITE : 0);
+        conn.setown(querySDS().connect(xpath.str(), myProcessSession(), mode, SDS_LOCK_TIMEOUT));
+        if (!conn)
+            return false;
+        xpath.set("//" MSG_NODE);
+        if (filter->queryHiddenOnly())
+            xpath.append("[" ATTR_HIDDEN "='1')]");
+        if (filter->queryVisibleOnly())
+            xpath.append("[" ATTR_HIDDEN "='0')]");
+        if (!isEmptyString(filter->queryMatchSource()))
+            xpath.appendf("[" ATTR_SOURCE "='%s']", filter->queryMatchSource());
+        if (filter->queryMatchCode())
+            xpath.appendf("[" ATTR_CODE "='%d']", (int)filter->queryMatchCode());
+        if (filter->queryMatchAudience()!=MSGAUD_all)
+            xpath.appendf("[" ATTR_AUDIENCE "='%s']", LogMsgAudienceToFixString(filter->queryMatchAudience()));
+        if (filter->queryMatchClass()!=MSGCLS_all)
+            xpath.appendf("[" ATTR_CLASS "='%s']", LogMsgClassToFixString(filter->queryMatchClass()));
+        if (filter->queryMatchMsgId())
+            xpath.appendf("[" ATTR_MSGID "='%" I64F "u']", filter->queryMatchMsgId());
+        if (filter->queryMatchTimeStamp())
+            xpath.appendf("[" ATTR_TIMESTAMP "='%" I64F "u']", filter->queryMatchTimeStamp());
+        root.setown(conn->getRoot());
+        msgIter.setown(root->getElements(xpath.str()));
+        msgIter->first();
+        return ensureMatch();
+    }
+    virtual bool next() override
+    {
+        msgIter->next();
+        return ensureMatch();
+
+    }
+    virtual bool isValid() override
+    {
+        return msgIter ? msgIter->isValid() : false;
+    }
+};
+
+ISysInfoLoggerMsgIterator * createSysInfoLoggerMsgIterator(bool visibleOnly, bool hiddenOnly, unsigned year, unsigned month, unsigned day)
+{
+    Owned<CSysInfoLoggerMsgFilter> filter = new CSysInfoLoggerMsgFilter(visibleOnly, hiddenOnly, year, month, day);
+    return new CSysInfoLoggerMsgIterator(filter.getClear(), false);
+}
+
+ISysInfoLoggerMsgIterator * createSysInfoLoggerMsgIterator(unsigned __int64 msgId, unsigned __int64 ts)
+{
+    Owned<CSysInfoLoggerMsgFilter> filter = new CSysInfoLoggerMsgFilter(msgId, ts);
+    return new CSysInfoLoggerMsgIterator(filter.getClear(), false);
+}
+
+ISysInfoLoggerMsgIterator * createSysInfoLoggerMsgIterator(ISysInfoLoggerMsgFilter * msgFilter)
+{
+    return new CSysInfoLoggerMsgIterator(msgFilter, false);
+}
+
+void logSysInfoError(const LogMsgId msgId, const LogMsgCategory & cat, LogMsgCode code, const char *source, const char * msg, unsigned __int64 ts)
```

Comment:
I'm not sure what/where msgId is to come from(?) - I don't think it should be related to logId of jlog logs - which can reset/overlap per process.
It may make sense for each entry to have an id of some kind, but not sure one should be passed in.
It would make sense for this mechanism to attribute some kind of unique id to each syslog message, but need to think carefully how they can be accessed efficiently (it would be useful so the ID's can be shared as an access index).

I am thinking that the id should be in effect the access time+counter, e.g. "m202405d24-99"
That way you can use the ID to quickly access /SysLogs/m202405/d24/msg99

Let's discuss further. It implies a slight change to the format.

Reply:
I have tried using the message node in the form: msg##.  Unfortunately, the iterator crashes with "Illegal instruction", if I attempt to use wildcards with child search.  E.g. this fails: //msg*[@hidden='0']

I'm not sure if it's the wildcard with child search or wildcard with attribute that is causing the problem. 
 I've created a jira for this issue: https://hpccsystems.atlassian.net/browse/HPCC-31960